wled00/wled.cpp

```diff
@@ -437,14 +437,64 @@ void WLED::initConnection()
   // Only initialize ethernet board if not NONE
   if (ethernetType != WLED_ETH_NONE && ethernetType < WLED_NUM_ETH_TYPES) {
     ethernet_settings es = ethernetBoards[ethernetType];
-    ETH.begin(
-      (uint8_t) es.eth_address, 
-      (int)     es.eth_power, 
-      (int)     es.eth_mdc, 
-      (int)     es.eth_mdio, 
-      (eth_phy_type_t)   es.eth_type,
-      (eth_clock_mode_t) es.eth_clk_mode
-    );
+    // Use PinManager to ensure pins are available for
+    // ethernet AND to prevent other uses of these pins.
+    bool s = true;
+    byte pinsAllocated[4] { 255, 255, 255, 255 };
+
+    if (s && allocatePin((byte)es.eth_address)) {
+      pinsAllocated[0] = (byte)es.eth_address;
+    } else {
+      s = false;
+    }
+    if (s && allocatePin((byte)es.eth_power)) {
+      pinsAllocated[1] = (byte)es.eth_power;
+    } else {
+      s = false;
+    }
+    if (s && allocatePin((byte)es.mdc)) {
+      pinsAllocated[2] = (byte)es.eth_mdc;
+    } else {
+      s = false;
+    }
+    if (s && allocatePin((byte)es.mdio)) {
+      pinsAllocated[2] = (byte)es.eth_mdio;
+    } else {
+      s = false;
+    }
+    switch(es.eth_clk_mode) {
+      ETH_CLOCK_GPIO0_IN:
+        s = allocatePin(0, false);
+        break;
+      ETH_CLOCK_GPIO0_OUT:
+        s = allocatePin(0);
+        break;
+      ETH_CLOCK_GPIO16_OUT:
+        s = allocatePin(16);
+        break;
+      ETH_CLOCK_GPIO17_OUT:
+        s = allocatePin(16);
+        break;
+      default:
+        s = false;
+        break;
+    }
+
```

Comment:
You can use logic `if (s) {s &= pinManager.allocatePin(x)); ...}` for a shorter notation.

Reply:
I think I would need to short-circuit the evaluation, so it would end up being:

```C++
if (s && (s = pinManager.allocatePin(x))) {
    pinsAllocated[y] = x;
}
```

I feel assignment statements within `if` conditions may be considered an arcane coding style.  I prefer to spend a couple more lines of code, if it makes the code easier to grok for later maintainers, since it compiles out the same. 

However, to smooth the review, I've made this change in 497c605.